yoti-sdk-impl/src/test/java/com/yoti/api/client/spi/remote/call/docs/DocScanServiceTest.java

```diff
@@ -0,0 +1,867 @@
+        } catch (DocScanException ex) {
+            assertSame(ex.getCause(), ioException);
+            assertThat(ex.getMessage(), containsString("Error building the request: Some io exception"));
+            return;
+        }
+        fail("Expected an exception");
+    }
+
+    @Test
+    public void createSession_shouldWrapURISyntaxException() throws Exception {
+        URISyntaxException uriSyntaxException = new URISyntaxException("someUrl", "Failed to build URI");
+
+        SessionSpec sessionSpecMock = mock(SessionSpec.class);
+        doReturn(signedRequestBuilderMock).when(docScanService).getSignedRequestBuilder();
+        when(signedRequestBuilderMock.build()).thenThrow(uriSyntaxException);
+
+        try {
+            docScanService.createSession(SOME_APP_ID, KEY_PAIR, sessionSpecMock);
+        } catch (DocScanException ex) {
+            assertSame(ex.getCause(), uriSyntaxException);
+            assertThat(ex.getMessage(), containsString("Error building the request: Failed to build URI: someUrl"));
+            return;
+        }
+        fail("Expected an exception");
+    }
+
+    @Test
+    public void createSession_shouldWrapGeneralException() throws Exception {
+        final Exception someException = new Exception("Some exception we weren't expecting");
+
+        SessionSpec sessionSpecMock = mock(SessionSpec.class);
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock i) throws Throwable {
+                throw someException;
+            }
+        }).when(docScanService).getSignedRequestBuilder();
+
+        try {
+            docScanService.createSession(SOME_APP_ID, KEY_PAIR, sessionSpecMock);
+        } catch (DocScanException ex) {
+            assertSame(ex.getCause(), someException);
+            assertThat(ex.getMessage(), containsString("Error creating the session: Some exception we weren't expecting"));
+            return;
+        }
+        fail("Expected an exception");
+    }
+
+    @Test
+    public void createSession_shouldCallSignedRequestBuilderWithCorrectMethods() throws Exception {
+        SessionSpec sessionSpecMock = mock(SessionSpec.class);
+        SimpleCreateSessionResult simpleCreateSessionResultMock = mock(SimpleCreateSessionResult.class);
+
+        doReturn(signedRequestBuilderMock).when(docScanService).getSignedRequestBuilder();
+        when(objectMapperMock.writeValueAsBytes(sessionSpecMock)).thenReturn(SOME_SESSION_SPEC_BYTES);
+        when(signedRequestBuilderMock.build()).thenReturn(signedRequestMock);
+        when(signedRequestMock.execute(SimpleCreateSessionResult.class)).thenReturn(simpleCreateSessionResultMock);
+        when(unsignedPathFactoryMock.createNewYotiDocsSessionPath(SOME_APP_ID)).thenReturn(SOME_PATH);
+
+        CreateSessionResult result = docScanService.createSession(SOME_APP_ID, KEY_PAIR, sessionSpecMock);
+
+        assertThat(result, Matchers.<CreateSessionResult>is(simpleCreateSessionResultMock));
+
```

Comment:
Not sure if they should be here or somewhere else, but can we add tests for the CreateSession getters: `getClientSessionTokenTtl`, `getClientSessionToken` and `getSessionId`?